lib/client-assets.php

```diff
@@ -583,7 +583,11 @@ function gutenberg_preload_api_request( $memo, $path ) {
 			$data['_links'] = $links;
 		}
 
-		$memo[ $path ] = array(
+		if ( 'OPTIONS' === $args['method'] ) {
+			$response = rest_send_allow_header( $response, $server, $request );
```

Comment:
Is the above `rest_do_request` assignment to `$response` wasted effort for `OPTIONS` request?

Reply:
Yes it's needed since `rest_do_request` is using `WP_REST_Server->dispatch()` and not `WP_REST_Server->serve_request()` so the needed headers for the request are never set in `dispatch` they are set in `serve_request`.